### DISCUSSION TITLE
Error: Script execution timed out after 1000ms

### DISCUSSION THREAD

**Discussion by fernandocalsa** [Budibase/budibase#8293] [Help] [answered]
Hello!

I've created an automation, and I have a JS script step. It looks like it takes some time to finish the job (is handling a lot of data so I think is ok to spend some time for this script to finish)

When the automation runs it fails in this step with this error

```
Error: Script execution timed out after 1000ms
```

I've been looking for the error in the repo but I couldn't find it. Do you know if this is thrown by some other library? Is there an environment variable to increase the time limit?

Budibase version: 2.0.31
deployed in kubernetes

**Comment by avivkri** [comment]
Hello @fernandocalsa,

Based on the discussions here https://github.com/Budibase/budibase/pull/5267 and https://github.com/Budibase/budibase/pull/5267/commits/09694e0045e9a3bb085c87da243042f5afb202ce:

In Kubernetes based installation, I used this environment variable in **app-service deployment** to bump up the automation step timeout to 5 minutes.

```
env:
- name: QUERY_THREAD_TIMEOUT
  value: "300000"
```

I hope you find this helpful.

**Comment by andz-bb** [comment, marked as answer]
hey @fernandocalsa - this value is currently hardcoded here as `JS_TIMEOUT_MS`: https://github.com/Budibase/budibase/blob/develop/packages/server/src/utilities/scriptRunner.js

Out of curiosity, what kind of volume of data are you working with? I think you should be OK to bump up the timeout limit here if you're running it self hosted.

**Comment by Hedronmx** [comment]
I am running onto the same issue... Any help on how can I increase the default timeout?

**Reply by andz-bb** [reply]
hey @Hedronmx, is your Budibase self-hosted or is it on cloud? What is the specific error you're seeing? Thanks

**Reply by Hedronmx** [reply]
Self hosted on a digital ocean droplet. I want to increase the query timeout. I deal with big data sets and sometimes the respond takes over 10 seconds to reply.

I already tried adding QUERY_THREAD_TIMEOUT: 55000 to the services env in my docker compose file but I still get 10 seconds time out.

Thanks for replying to this dead thread

**Reply by xnohat** [reply]
They still hardcode it https://github.com/Budibase/budibase/blob/develop/packages/server/src/utilities/scriptRunner.ts
If you use selfhosted, you can manually change it inside docker container name "app-service"

**Reply by Hedronmx** [reply]
Would that be inside the docker compose? o the image itself of the docker container? Thanks for your time, and sorry if its a dumb question. Im new to docker
